lib/models/personal_profile.dart

```diff
@@ -72,13 +81,15 @@ class SupporterMetadata {
   factory SupporterMetadata.fromMap(Map map) {
     final type = map["\$type"];
     return switch (type) {
-      "patreon" => PatreonSupporter.fromMap(map),
-      _ => SupporterMetadata(),
+      "patreon" => SubscriptionSupporter.fromMap(map),
+      "stripe" => SubscriptionSupporter.fromMap(map),
+      _ => SupporterMetadata()
     };
   }
 }
 
-class PatreonSupporter extends SupporterMetadata {
+class SubscriptionSupporter extends SupporterMetadata {
+  final String type;
```

Comment:
`$type` shouldn't really be included in the actual model, it's just a metadata field that identifies the type of object that is sent IIRC, and ideally should just be used to decide what model to use for deserializing on the client-side.

Reply:
Having it was mainly to avoid the extra PatreonSupporter class on top of SubscriptionSupporter since all of the api fields are shared between patreon and stripe using SubscriptionSupporter. Can still make those if you want.

Reply:
I see, I'd either make empty PatreonSupporter and StripeSupporter classes that derive from SubscriptionSupporter without adding any fields, or keep the type field but make it an enum instead of a string. Either works for me, it's up to you.